### PR TITLE
feat: sharpen image

### LINKED ISSUE #286: Sharpen image
### Is your feature request related to a problem?

I'd like to be able to sharpen an image.

### Desired solution

Add a method `sharpen` to `Image` that returns the sharpened image and leaves the original unchanged.

### Possible alternatives (optional)

_No response_

### Screenshots (optional)

_No response_

### Additional Context (optional)

_No response_

### PR DESCRIPTION
**Pull request by patrikguempel**
Closes #286.

### Summary of Changes
Added sharpening feature for images.

Co-authored-by: daniaHu <129186516+daniaHu@users.noreply.github.com>

### CODE REVIEW

**Comment by lars-reimann** [comment]
## [🦙 MegaLinter](https://megalinter.io/7.0.4) status: ❌ [ERROR](https://github.com/Safe-DS/Stdlib/actions/runs/5289620708)

| Descriptor  |                                Linter                                 |Files|Fixed|                            Errors                            |Elapsed time|
|-------------|-----------------------------------------------------------------------|-----|----:|--------------------------------------------------------------|------------|
|✅ PYTHON    |[black](https://megalinter.io/7.0.4/descriptors/python_black)          |    2|    1|                                                             0|0.86s       |
|❌ PYTHON    |[mypy](https://megalinter.io/7.0.4/descriptors/python_mypy)            |    2|     |[1](https://github.com/Safe-DS/Stdlib/actions/runs/5289620708)|1.87s       |
|✅ PYTHON    |[ruff](https://megalinter.io/7.0.4/descriptors/python_ruff)            |    2|    2|                                                             0|0.02s       |
|✅ REPOSITORY|[git_diff](https://megalinter.io/7.0.4/descriptors/repository_git_diff)|yes  |     |no                                                            |0.01s       |

See detailed report in [MegaLinter reports](https://github.com/Safe-DS/Stdlib/actions/runs/5289620708)
_Set `VALIDATE_ALL_CODEBASE: true` in mega-linter.yml to validate all sources, not only the diff_

_MegaLinter is graciously provided by [OX Security](https://www.ox.security/?ref=megalinter)_

File: tests/safeds/data/image/containers/test_image.py [hunk 1]
```diff
@@ -247,3 +247,19 @@ def test_should_be_original(self) -> None:
         image = Image.from_png_file(resolve_resource_path("image/original.png"))
         image2 = image.flip_horizontally().flip_horizontally()
         assert image == image2
+
+
+class TestSharpen:
+    @pytest.mark.parametrize("factor", [2, 25])
+    def test_should_sharpen(self, factor: int) -> None:
```
Comment (alex-senger) [inline tests/safeds/data/image/containers/test_image.py:254]:
```suggestion
    def test_should_sharpen(self, factor: float) -> None:
```

**Review by alex-senger — DISMISSED** [review]
[LGTM](https://lgtm.party/p/ABFI2HRO/)

**Review by jxnior01 — APPROVED** [review]
LGTM!!

**Review by alex-senger — APPROVED** [review]
[LGTM](https://lgtm.party/p/1TSSJAG/)

**Review by PhilipGutberlet — APPROVED** [review]
good work

**Pull request merged**